spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpOutboundGateway.java

```diff
@@ -61,19 +61,25 @@
 public class TcpOutboundGateway extends AbstractReplyProducingMessageHandler
 		implements TcpSender, TcpListener, Lifecycle {
 
-	private volatile AbstractClientConnectionFactory connectionFactory;
+	private static final int DEFAULT_SECOND_CHANCE_DELAY = 2;
```

Comment:
You don't use this constant BTW.
I mean that `secondChanceDelay` is really without those 2 seconds value...
Fixing on merge...